core/src/main/java/org/sonarsource/sonarlint/core/client/api/common/ModuleInfo.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.client.api.common;
+
+public class ModuleInfo {
```

Comment:
Yes it would, but this also means make engine classes generic. Might require more work. As discussed let's see at the end of the sprint when everything is settled if we want to make the change.